Automattic/jetpack
# Payments: add a default scenario for our upgrade nudges

Fixes #31982

## Proposed changes:

We currently have the following logic for the support doc link in the Stripe connection nudge banner:

**For the Payments buttons:**

- on WordPress.com Simple sites: https://wordpress.com/support/wordpress-editor/blocks/payments/
- On self-hosted sites: https://jetpack.com/support/jetpack-blocks/payments-block/#related-fees

**For the Donations block:**

- on simple: https://wordpress.com/support/wordpress-editor/blocks/donations/
- on self-hosted: https://jetpack.com/support/jetpack-blocks/donations-block/

For the Premium Content block **as well as any other block (this ends up being the default):**

- on simple: https://wordpress.com/support/wordpress-editor/blocks/premium-content-block/ -> this currently redirects to https://wordpress.com/support/wordpress-editor/blocks/paid-content-block/
- on self-hosted: https://jetpack.com/support/jetpack-blocks/payments-block/#related-fees

That last, default behavior seems to be the problem here, because we have at least one other block name, `recurring-payments`, that is not handled here. This PR fixes that by setting up a default.

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

This can be tested on WordPress.com Simple and on self-hosted, on a site where you have not tested those blocks before and have not connected your site to a Stripe account. In that scenario:

1. Go to Posts > Add New
2. Add a Payments block, and then pick each variation, one at a time, and check the support link in the Stripe connection banner.
